File: 02. 구현/#1. 왕실의 나이트.py
#1. 왕실의 나이트 - p. 115

input_data = input()
row = int(input_data[1])
columns = int(int(ord(input_data[0])) - ord('a') + 1)

steps = [(1, 2), (1, -2), (-1, 2), (-1, -2), (2, 1), (2, -1), (-2, 1), (-2, -1)]
count = 0

for step in steps:
    next_row = row + step[0]
    next_columns = columns + step[1]
    if next_row<1 or next_columns<1 or next_row>8 or next_columns>8:
        continue
    
    count += 1
    
print(count)

File: 02. 구현/#1. 왕실의 나이트.swift
import Foundation

let input: String = readLine()!
let row: Int = Int(String(input[input.index(after: input.startIndex)]))!
let column: Int = Int(input[input.startIndex].asciiValue!)
                                    - Int(Character("a").asciiValue!) + 1

let steps: [(x: Int, y: Int)] = [
    (1, 2), (-1, 2), (1, -2), (-1, -2), (2, 1), (-2, 1), (2, -1), (-2, -1)
]

var count: Int = 0
for step in steps {
    var nextPosition: (x: Int, y: Int) = (row+step.x, column+step.y)
    
    if nextPosition.x < 1 || nextPosition.y < 1 || nextPosition.x > 8 || nextPosition.y > 8 {
        continue
    }
    
    count += 1
}

print(count)
